umbraco-cloud/set-up/project-settings/umbraco-cicd/samplecicdpipeline/github-actions.md

```diff
@@ -1,71 +1,238 @@
+
+{% tabs %}
+{% tab title="Powershell" %}
+
+For a pipeline that uses Powershell scripts you will need the following files:
+- From the root folder
+  - `cloud.zipignore`
+- From the `powershell` folder
+  - `Get-LatestDeployment.ps1`
+  - `Get-ChangesById.ps1`
+  - `New-Deployment.ps1`
+  - `Add-DeploymentPackage.ps1`
+  - `Start-Deployment.ps1`
+  - `Test-DeploymentStatus.ps1`
+- From the `powershell/github` folder
+  - `main.yml`
+  - `cloud-sync.yml` 
+  - `cloud-deployment.yml`
+
+**Do the following to prepare the pipeline:**
+- Copy the `cloud.zipignore` file to the root of your repository
+- Make a copy of the `.gitignore` from your repository and call the copy `cloud.gitignore`
+  - Both files should be in the root of your repository
+  - In the bottom of the `.gitignore` file add the line `**/git-patch.diff`
+- Also in the root, create a folder called `.github`
+- Inside `.github` create two additional folders
+  - `workflows`
+  - `powershell`
+- Copy the 3 YAML files from the `github` folder into the `workflows` folder
+- Copy the Powershell scripts from the `powershell` folder to the `powershell` folder
+- Commit the all changes, and push to GitHub
+
+{% endtab %}
+
+{% tab title="Bash" %}
+
+For a pipeline that uses Bash scripts you will need the following files:
+- From the root folder
+  - `cloud.zipignore`
+- From the `bash` folder
+  - `get_latest_deployment.sh`
+  - `get_changes_by_id.sh`
+  - `create_deployment.sh`
+  - `upload_package.sh`
+  - `start_deployment.sh`
+  - `get_deployment_status.sh`
+- From the `bash/github` folder
+  - `main.yml`
+  - `cloud-sync.yml` 
+  - `cloud-deployment.yml`
+
+**Do the following to prepare the pipeline:**
+- Copy the `cloud.zipignore` file to the root of your repository
+- Make a copy of the `.gitignore` from your repository and call the copy `cloud.gitignore`
+  - Both files should be in the root of your repository
+  - In the bottom of the `.gitignore` file add the line `**/git-patch.diff`
+- Also in the root, create a folder called `.github`
+- Inside `.github` create two additional folders
+  - `workflows`
+  - `scripts`
+- Copy the 3 YAML files from the `github` folder into the `workflows` folder
+- Copy the Bash scripts from the `bash` folder to the `scripts` folder
+- Commit the all changes, and push to GitHub
+
+{% endtab %}
+{% endtabs %}
+
+The push will start a new pipeline run.
+
+{% hint style="info" %}
+
+Download the provided sample scripts as ZIP from the [GitHub repository](https://github.com/umbraco/Umbraco.Cloud.CICDFlow.Samples/tree/main). Click on "Code" and then choose "Download ZIP". Then unzip it and use those files for the next steps.
+
+{% endhint %}
+
+### Optional: Run a deployment to confirm
+
+With everything set up, you may want to confirm that Umbraco Cloud reflects the changes you are sending via your pipeline.
+
+While working on you project locally, add a new Document type.
+- Commit the change to `main` and push to your repository.
```

Comment:
shouldn´t this be changed as "- Commit the change to `main` (or `master` if you did not change the branch name) in Branch and push to your repository."?

Reply:
also, unlike azure devops, if the branch was not changes to main and is still master, there were no other extra steps like changing some yaml files to update branch name right?

Reply:
GitHub also has the section about changing master branch to mail - optionally